core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -400,19 +401,25 @@ static struct tee_mmap_region *init_xlation_table(struct tee_mmap_region *mm,
 
 		if (desc == UNSET_DESC) {
 			/* Area not covered by a region so need finer table */
-			uint64_t *new_table = xlat_tables[next_xlat++];
+			uint64_t *xlat;
 
-			/* Clear table before use */
-			DMSG("xlat used: %d/%d", next_xlat, MAX_XLAT_TABLES);
-			if (next_xlat > MAX_XLAT_TABLES)
-				panic("running out of xlat tables");
-			memset(new_table, 0, XLAT_TABLE_SIZE);
+			if (*table) {
+				assert(DESC_ATTRS(*table) == TABLE_DESC);
+				xlat = phys_to_virt(*table &
+						    ~DESC_ATTRS(*table),
+						    MEM_AREA_TEE_RAM_RW);
+			} else {
+				DMSG("xlat %d/%d", next_xlat, MAX_XLAT_TABLES);
+				if (next_xlat >= MAX_XLAT_TABLES)
+					panic("running out of xlat tables");
+				xlat = xlat_tables[next_xlat++];
+				memset(xlat, 0, XLAT_TABLE_SIZE);
+			}
 
-			desc = TABLE_DESC | virt_to_phys(new_table);
+			desc = TABLE_DESC | virt_to_phys(xlat);
 
 			/* Recurse to fill in new table */
-			mm = init_xlation_table(mm, base_va, new_table,
-					   level + 1);
+			mm = init_xlation_table(mm, base_va, xlat, level + 1);
```

Comment:
This is crappy. If a mm reference is leaves room for the next mm in the xlat, the increment `table++` below craps the later reuse of same xlat table.